documentation/docs/decentralized_identifiers/resolve.mdx

```diff
@@ -7,18 +7,152 @@ keywords:
 - Resolve
 ---
 import CodeSnippet from '../../src/components/CodeSnippetComponent'
-import resolution from  '!!raw-loader!../../../examples/low-level-api/resolution.rs';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
 
-TODO: Explain how resolving works including arguments.
 
-## Example
+DID resolution is the process of fetching a [DID Document](https://www.w3.org/TR/did-core/#dfn-did-documents) corresponding to a given [DID](https://www.w3.org/TR/did-core/#dfn-decentralized-identifiers). 
+The [IOTA Identity Framework](https://github.com/iotaledger/identity.rs) supports resolving DID Documents that are stored on an IOTA Tangle (public or private). The main tool supplied 
+by the IOTA Identity Framework to handle DID Document resolution in a type safe manner is the `Resolver`. A DID Resolver as defined in the [W3C Decentralized Identifiers specification](https://www.w3.org/TR/did-core/#dfn-did-resolvers)
+enforces the signature of the resolution function in a manner that is more centered around Web/API resolution rather than a strongly typed framework. This is the reason why the `Resolver` provided by the IOTA Identity Framework deviates somewhat from 
+the W3C specification.
 
-A short example that shows how you can resolve a DID Document. This example returns the latest DID Document.
 
-<CodeSnippet 
```

Comment:
Please keep the example. You can move it to the bottom if you don't want to refer to it at all.

Reply:
That example uses the now deprecated spec compliant resolution API hence I think it is better if we no longer link to it in our v0.5 documentation. I have included code snippets explaining how to resolve using the new Resolver. Is that OK?

Reply:
TBH no, in that case please update the example to use the new resolution API ;D

Reply:
Ok, but that example should probably be a separate PR. I will attend to it now :+1:

Reply:
Is it better now?